src/apollo.js

```diff
@@ -1,8 +1,8 @@
 import 'isomorphic-fetch'
-import { createHttpLink } from 'apollo-link-http'
-import { InMemoryCache } from 'apollo-cache-inmemory'
 import ApolloClient from 'apollo-boost'
 
-const client = new ApolloClient();
+const client = new ApolloClient({
+	uri: (process.env.PROTOCOL || "http") + "://" + (process.env.HOST || "localhost") + (":" + (process.env.PORT || 80)) + "/graphql"
```

Comment:
Should probably default the port to 3000 here. The default port for `sapper dev` is 3000.

If someone cloned the repo and ran it, it wouldn't work with a default of 80.

Reply:
That's true, though it would be possible if `process.env.PORT` is set as well in the browser, as it is populated with 3000 anyway (server is working like this). Currently for some reason it is not possible to set these environment variables in the browser even with the `webpack.DefinePlugin` as described above. I'm not sure if this is a bug with sapper or I still haven't figured out yet how to set environment variables with webpack in the browser.

Reply:
any updates?

Reply:
Why don't use just use a relative URL?  `uri: "/graphql"`

Reply:
To answer your question, I thought it might sense to abstract this away into environment variables to be able to change this to an external server.

Though I'm currently not using svelte (anymore) so if anyone is interested in this, one should fork this.

Reply:
I actually have a PR pending in the Sapper repo that will let you use an internal server: https://github.com/sveltejs/sapper/pull/1215

Reply:
I actually have a PR pending in the Sapper repo that will let you specify a different server: https://github.com/sveltejs/sapper/pull/1215. The main use case I had in mind though is if you can refer to your backend by a different hostname on the server than the client

You need to be careful with specifying an external server though because then you need to deal with CORS. I don't think this code would work as is with an external server

Reply:
Yes, I would do it differently today indeed.
I did this PR, while checking out svelte+apollo.
I rather thought to use it locally with a different graphql server (also locally).
Production use should certainly look different (as you mentioned).